src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java

```diff
@@ -461,17 +489,17 @@ private static void checkConfig(final DB wsdb) throws WorkspaceCommunicationExce
 			}
 			return Optional.of(schemaVer);
 		} catch (MongoException me) {
-			throw new WorkspaceCommunicationException(
-					"There was a problem communicating with the database", me);
+			throw new WorkspaceCommunicationException(ERR_DB_COMM, me);
 		}
 	}
 	
-	static void ensureIndexes(final DB wsdb) throws CorruptWorkspaceDBException {
+	static void ensureIndexes(final MongoDatabase wsdb) throws CorruptWorkspaceDBException {
 		final HashMap<String, List<IndexSpecification>> indexes = getIndexSpecs();
 		for (final String col: indexes.keySet()) {
 			for (final IndexSpecification index: indexes.get(col)) {
 				try {
 					wsdb.getCollection(col).createIndex(index.index, index.options);
+					// this seems to be the 1 *&)&*)* place where DKE is still used
```

Comment:
hmmm, decoding `*&)&*)*` could be tricky for other developers... perhaps you should leave a hint as to how they can crack the code?